File: backend/app/schemas/admin_support.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class SupportTicketReplyCreate(BaseModel):
    reply_message: str = Field(min_length=1)


class SupportTicketAdminSummary(BaseModel):
    id: uuid.UUID
    subject: str
    category: str | None
    priority: str
    status: str
    requester_name: str
    requester_email: str
    created_at: datetime

    model_config = {"from_attributes": True}


class SupportTicketAdminDetail(BaseModel):
    id: uuid.UUID
    subject: str
    category: str | None
    priority: str
    status: str
    message: str
    requester_name: str
    requester_email: str
    reply_message: str | None
    replied_by_name: str | None
    replied_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/lib/types/admin-support.ts
// Mirrors backend/app/schemas/admin_support.py exactly — keep these two in sync.

export type SupportTicketStatus = "open" | "answered" | "closed";
export type SupportTicketPriority = "low" | "medium" | "high";

export interface SupportTicketReplyCreate {
  reply_message: string;
}

export interface SupportTicketAdminSummary {
  id: string;
  subject: string;
  category: string | null;
  priority: SupportTicketPriority;
  status: SupportTicketStatus;
  requester_name: string;
  requester_email: string;
  created_at: string;
}

export interface SupportTicketAdminDetail {
  id: string;
  subject: string;
  category: string | null;
  priority: SupportTicketPriority;
  status: SupportTicketStatus;
  message: string;
  requester_name: string;
  requester_email: string;
  reply_message: string | null;
  replied_by_name: string | null;
  replied_at: string | null;
  created_at: string;
}
